### ISSUE TITLE
Objectness loss calculation: cross entropy calculation

### ISSUE THREAD

**Issue by mpdroid** [ultralytics/yolov5#6544]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

In the objectness loss calculation `loss.py::compute_loss::__call__`  what is the purpose of the following statement?

```
tobj[b, a, gj, gi] = (1.0 - self.gr) + self.gr * score_iou  
```

`tobj` is used later to compute the BCE at;
```
obji = self.BCEobj(pi[..., 4], tobj) 
```
I see that `self.gr` (ground truth?)  is initialized to 1.0 in the constructor and never modified. So effectively `tobj` always equals to `score_iou`. Is there a hyperparameter that contols how `self.gr` is updated before calculating loss?

Ultimately, I am trying to understand how objectness is computed, and what levers are available to favor recall over precision.

**Thanks for creating and sharing this amazing body of work. Having the yolo code in python makes a big difference in comprehending how it works.*

**Comment by glenn-jocher** [comment]
@mpdroid gr is GIoU ratio. At the start of training box regression is very poor and IoU targets for objectness will be near zero, so this line provides space for warmup strategies here.

**Comment by mpdroid** [comment]
Thank you. Now I see that gr is initialized in warmup phase and commented out. I will look up GIoU ratio.
